src/UserSpaceInstrumentation/InstrumentProcess.cpp

```diff
@@ -0,0 +1,448 @@
+        if (trampoline_address_or_error.has_error()) {
+          LOG("Failed to allocate memory for trampoline: %s",
+              trampoline_address_or_error.error().message());
+          continue;
+        }
+        const uint64_t trampoline_address = trampoline_address_or_error.value();
+        TrampolineData trampoline_data;
+        trampoline_data.trampoline_address = trampoline_address;
+        OUTCOME_TRY(auto&& function_data, ReadTraceesMemory(pid_, function_address, backup_size));
+        trampoline_data.function_data = function_data;
+        auto address_after_prologue_or_error =
+            CreateTrampoline(pid_, function_address, function_data, trampoline_address,
+                             entry_payload_function_address_, return_trampoline_address_,
+                             capstone_handle, relocation_map_);
+        if (address_after_prologue_or_error.has_error()) {
+          LOG("Failed to create trampoline: %s", address_after_prologue_or_error.error().message());
+          OUTCOME_TRY(ReleaseMostRecentlyAllocatedTrampolineMemory(module_address_range));
+          continue;
+        }
+        trampoline_data.address_after_prologue = address_after_prologue_or_error.value();
+        trampoline_map_.emplace(function_address, trampoline_data);
+      }
+      auto it = trampoline_map_.find(function_address);
+      if (it == trampoline_map_.end()) {
+        continue;
+      }
+      const TrampolineData& trampoline_data = it->second;
+
+      auto result = InstrumentFunction(pid_, function_address, function_id,
+                                       trampoline_data.address_after_prologue,
+                                       trampoline_data.trampoline_address);
+      if (result.has_error()) {
+        LOG("Unable to instrument %s: %s", function.function_name(), result.error().message());
+      } else {
+        instrumented_function_ids.insert(function_id);
+      }
+    }
+  }
+
+  MoveInstructionPointersOutOfOverwrittenCode(pid_, relocation_map_);
+
+  OUTCOME_TRY(EnsureTrampolinesExecutable());
+
+  return instrumented_function_ids;
+}
+
+ErrorMessageOr<void> InstrumentedProcess::UninstrumentFunctions(
+    const CaptureOptions& capture_options) {
+  OUTCOME_TRY(AttachAndStopProcess(pid_));
+  orbit_base::unique_resource detach_on_exit{pid_, [](int32_t pid) {
+                                               if (DetachAndContinueProcess(pid).has_error()) {
+                                                 ERROR("Detaching from %i", pid);
+                                               }
+                                             }};
+
+  for (const auto& function : capture_options.instrumented_functions()) {
+    OUTCOME_TRY(auto&& modules, ModulesFromModulePath(function.file_path()));
+    for (const auto& module : modules) {
+      const uint64_t function_address =
+          module.address_start() + function.file_offset() - module.executable_segment_offset();
```

Comment:
if you store intrumented function addresses on `InstrumentFunctions` call you won't need to do this again.

Reply:
It will also protect you from user errors like accidentally providing altered/extended list of instrumented functions

Reply:
done